packages/@uppy/xhr-upload/src/index.ts

```diff
@@ -239,12 +239,9 @@ export default class XHRUpload<
           }
 
           const body = this.opts.getResponseData(res.responseText, res)
-          const uploadURL = body[this.opts.responseUrlFieldName]
-          if (typeof uploadURL !== 'string') {
-            throw new Error(
-              `The received response did not include a valid URL for key ${this.opts.responseUrlFieldName}`,
-            )
-          }
+          const uploadURL = body?.[this.opts.responseUrlFieldName] as
+            | string
+            | undefined
```

Comment:
```suggestion
          const uploadURL = body[this.opts.responseUrlFieldName]
          if (!(typeof uploadURL === 'string' || uploadURL == null)) {
            throw new Error(
              `The received response did not include a valid URL for option responseUrlFieldName ${this.opts.responseUrlFieldName}`,
            )
          }
```

Reply:
As you can read in the issue and my comment, the whole point is that this should _not_ be required.

Reply:
yes, and i think my suggested code is a safer way to check that rather than using a blind type assertion

Reply:
i'm not sure how my suggestion would make it the exact same as before this PR, as i added a check that allows `undefined` and `null` (`==` operator).

If `uploadURL` is `undefined`, `!(typeof uploadURL === 'string' || uploadURL == null)` evaluates to `false`, hence it will **not** throw the error.